src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -356,9 +362,16 @@ private void finishCurrentDeployment() throws InterruptedException {
 
     private void persistGroupToRootComponents(DeploymentDocument deploymentDocument) {
         Map<String, Object> deploymentGroupToRootPackages = new HashMap<>();
-        // TODO: [P41179087] Removal of group from the mappings. Currently there is no action taken
-        // when a device is removed from a thing group. Empty configuration is treated as a valid config
-        // for a group but not treated as removal.
+        Topics deploymentGroupTopics = config.lookupTopics(GROUP_TO_ROOT_COMPONENTS_TOPICS);
+        Topics groupMembershipTopics = config.lookupTopics(GROUP_MEMBERSHIP_TOPICS);
```

Comment:
Hmm, this is generally fine, but I can see edge cases where `persistGroupToRootComponents` does not get invoked in certain cases and will not get cleaned up and will cause the `GROUP_TO_ROOT_COMPONENTS_TOPICS` list to be inaccurate. Is there a better way to maintain this mapping instead of topics? Or maybe you could do `groupMembershipTopics.remove()` in DefaultDeploymentTask as well before recording the new list.

Reply:
Eventually we may need to persist this membership response from                    `ListThingGroupsForCoreDeviceRequest` even after deployment finishes

Reply:
Since GROUP_TO_ROOT_COMPONENTS_TOPICS is only updated after the deployment, group membership info need to be stored in topics to be persisted across kernel restarts. Added groupMembershipTopics.remove() in the DefaultDeploymentTask to make it more robust.

persistGroupToRootComponents is only called when the deployment succeeded or deployment failed with no rollback. Those are the two states when components on the device changes. When a deployment fails with no state change or deployment fails with rollback, the components on the device should not change.

Reply:
Sure, My concern was that the temporary mapping written to `GROUP_MEMBERSHIP_TOPICS` wouldn't get cleaned up always and adding `groupMembershipTopics.remove()` to DefaultDeploymentTask address it